File: C++/78_Subsets.cpp
class Solution {
public:
    vector<vector<int>> sol;
    vector<vector<int>> subsets(vector<int>& nums) {
        helper(nums, vector<int>{}, 0);
        return sol;
    }

    void helper(vector<int>&nums, vector<int> curr, int idx) {
        if(idx >= nums.size()) {
            sol.push_back(curr);
            return;
        }
        helper(nums, curr, idx + 1);
        curr.push_back(nums[idx]);
        helper(nums, curr, idx + 1);
    }
};

File: Python/78_Subsets.py
class Solution:

    def subsets(self, nums: List[int]) -> List[List[int]]:
        subsets = []
        def aux(curr, i):
            if i >= len(nums):
                subsets.append(curr)
                return
            else:
                aux( list(curr), i + 1)
                curr.append(nums[i])
                aux(list(curr), i + 1)
        aux([], 0)
        return subsets
